src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1674,6 +1675,10 @@ fn resolution_failure(
                     res.descr()
                 )
             };
+            let no_field_on_variant = |variant: Res| {
+                let variant = variant.name(tcx);
+                format!("variant `{variant}` has no such field")
+            };
```

Comment:
I'd rather not define a closure for this since it's pretty simple and only used once.